Arelle/Arelle
# NL BR KVK validations (2.04, 3.01, 4.07, 4.10, 4.16)

#### Description
BR-KVK-2.04: The period in the context MUST correspond to the period of the current or
previous financial reporting period, or one day before the start of the comparative financial
where the context must be instant.

Example:
Reporting Period Start Date: 2018-01-01
Reporting Period End Date: 2018-12-31

| Context | Instant    | Start Date | End Date   | Note                                                    |
| ------- | ---------- | ---------- | ---------- | ------------------------------------------------------- |
| 1       | 2018-12-31 |            |            | End value as context for the current reporting period   |
| 2       |            | 2018-12-31 | 2018-12-31 | Current reporting period                                |
| 3       | 2017-12-31 |            |            | End value as context for previous reporting period      |
| 4       |            | 2017-12-31 | 2017-12-31 | Previous reporting period                               |
| 5       | 2016-12-31 |            |            | Starting value as context for previous reporting period |


___
BR-KVK-3.01: A measure element with a namespace prefix that refers to the
"http://www.xbrl.org/2003/iso4217" namespace MUST appear exactly once in the instance document.

___
BR-KVK-4.07: The jenv-bw2-i:FinancialReportingPeriodCurrentEndDate MUST be before the date of filing.

___
BR-KVK-4.10: The jenv-bw2-i:DocumentAdoptionDate MUST NOT be after the date of filing.

___
BR-KVK-4.16: A corrected financial statement MUST be established.
If kvk-i:DocumentResubmissionDueToUnsurmountableInaccuracies is "Ja" (Yes),
the following facts must be filled for a corrected financial statement:
jenv-bw2-i:DocumentAdoptionStatus
jenv-bw2-i:DocumentAdoptionDate

#### Steps to Test
Run validation on each failure example document using NT16 disclosure system and confirm that the associated error is logged.

Run validation on `nt16.xbrl` and `nt17.xbrl` and confirm no NL validation plugin errors are logged.
___
**review**:
@Arelle/arelle
